OpenTrons/opentrons-api
# docs(api): use named arguments in code samples

# Overview

This PR is for RTC-32, Use Named Arguments in Code Samples. We've reviewed and inventoried all the code samples in the API docs. Most of the API methods in the tutorial use positional arguments. To help improve clarity for users, we're going to add the argument names to all the code samples.

The inventory is attached to the JIRA issue and linked here: [Named arguments](https://docs.google.com/document/d/1FE0jlKSh4OWpWMdrvU02Jyup4N8iiOd7xtpqwREyBgQ/edit#)

Opening as a draft PR. Will finish and push all revisions here before making this a real/regular PR and asking for reviews. 

This is a low-priority, time-permitting project. This PR may stay as a draft for a while.

# Test Plan

1. Format original code samples from the Opentrons API docs using the [Black Python code formatter](https://black.readthedocs.io/en/latest/).
2. Upload reformatted `.py` files in the Opentrons app for an automated check. If files pass the check, push to the draft PR.
3. After reviewing and pushing all the code, ask a willing engineer to review the reformatted code.
4. Make any recommended changes.
5. Change PR from a draft to regular PR and do the usual process. 

# Changelog

All the code samples in the API docs will change from using positional arguments to named arguments. Affects sample code examples in our online docs only.

Example
- Original code snippet: `tiprack = protocol.load_labware('opentrons_96_tiprack_300ul', 1)`
- Revised code snippet: `tiprack = protocol.load_labware(load_name= 'opentrons_96_tiprack_300ul', location= 1)`

# Review requests

Does this changed formatting look weird, not in line with standard practices, or incorrect?

**Sandbox URL**
http://sandbox.docs.opentrons.com/named-arguments-for-api-docs/v2/

# Risk assessment

Moderate. Even though this PR changes documentation code examples and snippets only, if revisions aren't accurate, code won't work for users attempting to learn how to use the Opentrons API. Code that doesn't work will contribute to user frustration and distrust with our documentation.
